### ISSUE TITLE
Error installing taskwarrior-web: 	ERROR: Failed to build gem native extension.

### ISSUE THREAD

**Issue by stef204** [theunraveler/taskwarrior-web#74]
Hi,
```
% gem install taskwarrior-web
```
results in:

```
/usr/bin/ruby -r ./siteconf20150724-19042-1435l57.rb extconf.rb
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling generator.c
In file included from /usr/include/stdio.h:27:0,
                 from /usr/include/ruby-2.2.0/ruby/defines.h:26,
                 from /usr/include/ruby-2.2.0/ruby/ruby.h:29,
                 from /usr/include/ruby-2.2.0/ruby.h:33,
                 from ../fbuffer/fbuffer.h:5,
                 from generator.c:1:
/usr/include/features.h:328:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
 #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
    ^
In file included from generator.c:1:0:
../fbuffer/fbuffer.h: In function ‘fbuffer_to_s’:
../fbuffer/fbuffer.h:175:47: error: macro "rb_str_new" requires 2 arguments, but only 1 given
     VALUE result = rb_str_new(FBUFFER_PAIR(fb));
                                               ^
../fbuffer/fbuffer.h:175:20: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
     VALUE result = rb_str_new(FBUFFER_PAIR(fb));
                    ^
Makefile:237: recipe for target 'generator.o' failed
make: *** [generator.o] Error 1

make failed, exit code 2
```
I am using: ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-linux]

**Comment by albertoaflores** [comment]
+1 - Same problem!

**Comment by denislobanov** [comment]
Reproduced on Arch Linux x86_64, same issue

**Comment by matthiasbeyer** [comment]
Reproduced on NixOS with both `ruby_2_2_2` and `ruby_2_2_0`. `ruby_2_0_0` worked.

**Comment by Pupo1** [comment]
Reproduced on Linux Mint 17.1 with ruby 1.9.3p484 (2013-11-22 revision 43786) [x86_64-linux]

**Comment by matthiasbeyer** [comment]
I consider this project dead now.

**Comment by Pupo1** [comment]
What a pity...

**Comment by alexgus** [comment]
I updated to json 1.8 and it works, I see the interface. Next issue -> no task displayed

**Comment by rmonico** [comment]
+1 reproduced on Manjaro Linux 15.12

Solved updating json to 1.8 too (thanks alexgus)

**Comment by mattdf** [comment]
JSON is broken again, but I've fixed it in my fork: https://github.com/mattdf/taskwarrior-web

**Comment by mattdf** [comment]
Should be fixed by #80
